src/main/java/uk/gov/hmcts/darts/dailylist/service/impl/DailyListProcessorImpl.java

```diff
@@ -0,0 +1,175 @@
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.common.entity.CourtCaseEntity;
+import uk.gov.hmcts.darts.common.entity.CourthouseEntity;
+import uk.gov.hmcts.darts.common.entity.DailyListEntity;
+import uk.gov.hmcts.darts.common.entity.HearingEntity;
+import uk.gov.hmcts.darts.common.entity.JudgeEntity;
+import uk.gov.hmcts.darts.common.exception.DartsApiException;
+import uk.gov.hmcts.darts.common.repository.HearingRepository;
+import uk.gov.hmcts.darts.common.service.RetrieveCoreObjectService;
+import uk.gov.hmcts.darts.courthouse.CourthouseRepository;
+import uk.gov.hmcts.darts.dailylist.enums.JobStatusType;
+import uk.gov.hmcts.darts.dailylist.enums.SourceType;
+import uk.gov.hmcts.darts.dailylist.exception.DailyListError;
+import uk.gov.hmcts.darts.dailylist.model.CitizenName;
+import uk.gov.hmcts.darts.dailylist.model.CourtList;
+import uk.gov.hmcts.darts.dailylist.model.DailyList;
+import uk.gov.hmcts.darts.dailylist.model.Defendant;
+import uk.gov.hmcts.darts.dailylist.model.Hearing;
+import uk.gov.hmcts.darts.dailylist.model.Sitting;
+import uk.gov.hmcts.darts.dailylist.repository.DailyListRepository;
+import uk.gov.hmcts.darts.dailylist.service.DailyListProcessor;
+
+import java.time.LocalDate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class DailyListProcessorImpl implements DailyListProcessor {
+    private final DailyListRepository dailyListRepository;
+    private final RetrieveCoreObjectService retrieveCoreObjectService;
+    private final CourthouseRepository courthouseRepository;
+    private final HearingRepository hearingRepository;
+
+
+    private final ObjectMapper objectMapper;
+
+    @Override
+    @Transactional
+    public void processAllDailyLists(LocalDate date) {
+        Arrays.stream(SourceType.values()).forEach(sourceType -> processDailyListForSourceType(date, sourceType));
+    }
+
+    private void processDailyListForSourceType(LocalDate date, SourceType sourceType) {
+        List<CourthouseEntity> allCourthouses = courthouseRepository.findAll();
+        for (CourthouseEntity allCourthouse : allCourthouses) {
+            List<DailyListEntity> dailyLists = dailyListRepository.findByCourthouse_IdAndStatusAndStartDateAndSourceOrderByPublishedTimestampDesc(
+                    allCourthouse.getId(),
+                    String.valueOf(JobStatusType.NEW),
+                    date, String.valueOf(sourceType)
+            );
+
+            // Daily lists are being ordered descending by date so first item will be the most recent version
+            if (!dailyLists.isEmpty()) {
+                try {
+                    processDailyList(dailyLists.get(0));
+                } catch (JsonProcessingException e) {
+                    throw new DartsApiException(DailyListError.FAILED_TO_PROCESS_DAILYLIST, e);
```

Comment:
looks like this might stop the processing of all the daily lists because 1 fails? 
We probably want to just mark this one as failed and carry on with the other courthouses

Reply:
ACCEPT.

Reply:
dont think so, but might need to ask @davet1985